kustomize/commands/localize/localize.go

```diff
@@ -62,6 +66,31 @@ kustomize localize https://github.com/kubernetes-sigs/kustomize//api/krusty/test
 			if err != nil {
 				return errors.Wrap(err)
 			}
+
+			if !f.noVerify {
+				currBuildOutput, err := exec.Command("kustomize", "build", args.target).Output()
+				if err != nil {
+					return errors.Wrap(err)
+				}
+
+				buildDst := dst
+				if f.scope != "" && f.scope != args.target {
+					buildDst = filepath.Join(dst, filepath.Base(args.target))
```

Comment:
question: Does this align with how the scope flag was built?

Reply:
That's what I thought too but if I didn't add this if block then the `TestScopeFlag` test failed with my implementation 🤔 I'll take another look at this code

Reply:
Here's the error message I currently get if I don't have this piece of code, if you have any insight 👀 
```
RUN   TestScopeFlag
    localize_test.go:63: 
                Error Trace:    /Users/sanaa/contributing/kustomize/kustomize/commands/localize/localize_test.go:63
                Error:          Received unexpected error:
                                unable to find one of 'kustomization.yaml', 'kustomization.yml' or 'Kustomization' in directory '/private/var/folders/_4/y164mr8934j6fzw5x_8v_ywr0000gn/T/kustomize-719637769/dst'
                Test:           TestScopeFlag
```

Reply:
Let me do some digging and I'll get back to you!

Reply:
thank you!! I appreciate all the helpful feedback you've been providing me on this PR

Reply:
This was a little tough, but I think I figured it out 😅

This specific test is using a target for `build`. The equivalent command line would be something to the effect of
```shell
kustomize build target/
```
where `target` is the folder that results from removing the scope path out of the target path in `kustomize localize scope/target newDir --scope scope`.

For that specific test case, the folder structure looks like this:
```console
.
|____target
| |____kustomization.yaml
|____base
  |____deployment.yaml
  |____kustomization.yaml
```
and the content of `target/kustomization.yaml` is:
```yaml
resources:
- ../base
```
which suggests `target` is the right folder to run `kustomize build` from.

For this to work, the code will need to take the presumed structure into account. I suspect considering `target` as the root folder where `kustomize build` would be run from is enough to make it work (and then the condition becomes superfluous).

Reply:
Thank you for the detailed breakdown! This clears things up but one questions I'd like to clarify is would you say that during the verify step for the localize command, for this test case we would only need to run `kustomize build` once? Since `target` is the right folder to run the build command from, ideally the output would be the same regardless?

Reply:
In my opinion the test itself shouldn't be aware how many times it's running `build`, since it aims to test the functionality in `kustomize localize`. In this specific instance, because the flag `--no-verify` is not in use, it should run build twice and pass verification.

Do you have any specific concern in mind? I suspect I'm not entirely following your thought process.

Reply:
I agree that it should run twice and the test shouldn't be aware. Let me break down what I'm thinking now that I have a better sense of it!

For this specific test where the structure is as what you outlined:
```
.
|____target
| |____kustomization.yaml
|____base
  |____deployment.yaml
  |____kustomization.yaml
```

the kustomize localize will look like `kustomize localize <scope>/target/ dst`

where the target will only the customization file but the destination file will have **both** the target folder and the base folder. 

In this case, when I run kustomize build, it needs to know which kustomization file to run on (whether that be target or base) but since its in the top level `dst` directory, it doesn't know where to look. What I'm trying to figure out is how do I go about adding some code to find which subdirectory to run build on! I hope that makes sense and if you have any further guidance on this 🙏

Reply:
Got it!
I think my rationale is a little tough to explain in words 😅 so please let me know if a diagram would be helpful.

Let's consider that the sample command in this test is equivalent to this:
```console
kustomize localize /path/to/test/folder/target localized/ --scope /path/to/test/folder
```

`/path/to/test/folder` has the structure we discussed before:
```console
.
|____target
| |____kustomization.yaml
|____base
  |____deployment.yaml
  |____kustomization.yaml
```

If we run the `kustomize localize` command from before, we'll be left with a folder structure of the format:
```console
.
|____target
| |____kustomization.yaml
|____localized
| |____target
| | |____kustomization.yaml
| |____base
|   |____deployment.yaml
|   |____kustomization.yaml
|____base
  |____deployment.yaml
  |____kustomization.yaml
```

In this case, my general rationale is that you can simply subtract the `scope` content from the `path` that was localized. In this example, this will leave you with `target/`, which you can append to `newDir` value (in this case `localized/`). This operation would leave you with the path `localized/target`, which is the correct place to run `kustomize build` from.

I think this scope use case is special in the sense that you can't run `kustomize build` directly from `target` without specifying a scope. However, for this to work, I suspect it might be needed to expand relative paths into absolute paths.

Reply:
Ah I see what you mean. In my original commit, I was building the scope path in order to run `kustomize build`. Based on this discussion, I'm going to add that back! Thanks!